server/metadata/tenant.go

```diff
@@ -166,6 +166,39 @@ func (m *TenantManager) CreateOrGetTenant(ctx context.Context, txMgr *transactio
 	return m.createOrGetTenantInternal(ctx, tx, namespace)
 }
 
+// CreateTenant is a thread safe implementation of creating a new tenant. It returns the error if it already exists.
+func (m *TenantManager) CreateTenant(ctx context.Context, txMgr *transaction.Manager, namespace Namespace) (tenant *Tenant, err error) {
+	m.Lock()
+	defer m.Unlock()
+
+	var ok bool
+	if tenant, ok = m.tenants[namespace.Name()]; ok {
+		if tenant != nil {
```

Comment:
if it is present in the map and the tenant is nil then it is a bug, use the tenant object directly. If it happens then it will help us to catch the bug.

Reply:
So this condition could be just: `if tenant = m.tenants[namespace.Name()]; tenant != nil {
// conflict`

Reply:
updated it.